core/storage/classifier/gae_models.py

```diff
@@ -111,3 +116,87 @@ def create(
 
         classifier_model_instance.put()
         return instance_id
+
+
+class TrainClassifierJobModel(base_models.BaseModel):
+    """Model for storing classifier training jobs.
+
+    The id of instances of this class has the form
+    {{exp_id}}.{{random_hash_of_16_chars}}
+    """
+
+    # The ID of the algorithm used to create the model.
+    algorithm_id = ndb.StringProperty(required=True, choices=ALGORITHM_CHOICES)
```

Comment:
Makes sense for algorithm_id and exp_id (which will never be updated). But, status is bound to get updated so indexing it _might_ not be the best solution. 

@seanlip why is it that equally query can't happen without indices? An index just speeds up the query after all.